File: FrontEnd/domain/entities/Invoice.ts
// Invoice domain entities — pure TypeScript, no framework dependencies

export interface LineItem {
  description: string;
  quantity: number;
  rate: number;
  amount: number;
}

export interface Invoice {
  id: string;
  invoiceNumber?: string;
  invoice_number?: string;
  hotel: string;
  hotel_id?: string;
  hotel_name?: string;
  amount?: number;
  baseAmount?: number;
  gst?: number;
  total?: number;
  period?: string;
  period_start?: string;
  period_end?: string;
  status: 'Paid' | 'Overdue' | 'Pending' | 'Draft' | 'paid' | 'pending' | 'overdue' | 'failed' | string;
  dueDate: string;
  due_date?: string;
  issuedDate?: string; // Optional in entities, required in types?
  generated_on?: string;
  daysOverdue?: number;
  items?: LineItem[];
}

File: BackEnd/app/schemas/invoice.py
from uuid import UUID
from pydantic import BaseModel
from typing import Optional


class InvoiceBase(BaseModel):
    amount: float
    status: Optional[str] = "Pending"
    period_start: str
    period_end: str
    generated_on: Optional[str] = None
    due_date: str


class InvoiceCreate(InvoiceBase):
    hotel_id: UUID


class InvoiceUpdate(BaseModel):
    status: Optional[str] = None
    amount: Optional[float] = None
    due_date: Optional[str] = None


class Invoice(InvoiceBase):
    id: UUID
    hotel_id: UUID
    hotel_name: Optional[str] = None
    invoice_number: Optional[str] = None

    class Config:
        from_attributes = True
